src/SCTPWrapper.cpp

```diff
@@ -98,6 +98,42 @@ void SCTPWrapper::OnNotification(union sctp_notification *notify, size_t len) {
       break;
     case SCTP_STREAM_RESET_EVENT:
       SPDLOG_TRACE(logger, "OnNotification(type=SCTP_STREAM_RESET_EVENT)");
+      struct sctp_stream_reset_event reset_event;
+      reset_event = notify->sn_strreset_event;
+      for (int i = 1; i < 2; i++) {
+        uint16_t streamid = reset_event.strreset_stream_list[i];
+        uint16_t set_flags;
+        if (reset_event.strreset_flags != 0) {
+          if ((reset_event.strreset_flags ^ SCTP_STREAM_RESET_INCOMING_SSN) == 0) {
+            set_flags = SCTP_STREAM_RESET_INCOMING;  
+          }
+          if ((reset_event.strreset_flags ^ SCTP_STREAM_RESET_OUTGOING_SSN) == 0) {
+            //fires when we close the stream from our side explicity or
+            //as a result of remote close or some error.
+            
+            logger->info("Outgoing stream_id#{} have been reset, calling onClose CB", streamid);
+            const uint8_t dc_close_data = DC_TYPE_CLOSE;
+            const uint8_t *dc_close_ptr = &dc_close_data;
+            OnMsgReceived(dc_close_ptr, sizeof(dc_close_ptr), streamid + 1, PPID_CONTROL);
```

Comment:
The data channel that's created had the stream id as "1" while the event I get here always seems to be having a stream id of "0". With that observation, I was trying to close our data channel (stream id 1) with the assumption that the corresponding stream id from the event side would be 0 as in this case. Obviously I need more clarity here and shouldn't be doing this.